finagle/h2/src/test/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4StreamTransportTest.scala

```diff
@@ -510,7 +499,7 @@ class Netty4StreamTransportTest extends FunSuite {
     assert(rspStreamQ.offer(Netty4Message.Trailers(trailers)))
     ctx.synchronized {
       assert(written.head == new DefaultHttp2HeadersFrame(trailers, true)
-        .stream(H2FrameStream(id, Http2Stream.State.OPEN)))
+        .stream(H2FrameStream(id, Http2Stream.State.HALF_CLOSED_LOCAL)))
```

Comment:
We now change the stream state _before_ writing the frame.  I think that's okay???